### PR TITLE
Add support for themeable favicons

### LINKED ISSUE #1347: Allow setting favicon in a theme
**Is your feature request related to a problem? Please describe.**

I'd like to be able to specify a custom favicon image from within a theme. This image would override the default DSpace favicon.

**Describe the solution you'd like**

I expect there would be two steps to this:
1. Place the custom favicon.ico image in the /images directory of the active theme
2. Specify the path to the new favicon image in a way that it replaces the default favicon path

Step 2 could be omitted if an automated step were added which looked for a file named favicon.ico in the active theme and either replaced the default favicon image or updated the favicon path to point to this image.

**Describe alternatives or workarounds you've considered**

The current work-around is to place the favicon image in the theme, then update the top-level index.html value to change:
```
<link rel="icon" type="image/x-icon" href="assets/images/favicon.ico" />
```
to:
```
<link rel="icon" type="image/x-icon" href="/assets/<my-theme>/images/favicon.ico" />
```

This method is effective, it simply breaks the theme model by requiring changes to the look of a site to be made outside of the active theme.

**Additional context**

### PR DESCRIPTION
**Pull request by bruno-atmire**
## References
* Fixes #1347 

## Description
Set the favicon (and other tags in the head section) based on the theme that's currently active.

## Instructions for Reviewers
Icons are picked in following order:
* head tags configured on the currently active theme (if any)
* head tags of the parent theme (if any) (recursive)
* head tags of the default theme (if any)
* hardcoded favicon: `src/assets/images/favicon.ico`

The favicon of the `dspace` theme is set to the logo of DSpace OOTB. The hardcoded favicon is also the DSpace logo.

I created a branch for testing: https://github.com/atmire/dspace-angular/tree/w2p-85123_theme-specific-favicon-review.
See [environment.common.ts](https://github.com/atmire/dspace-angular/blob/w2p-85123_theme-specific-favicon-review/src/environments/environment.common.ts#L293-L431) for the configured themes and favicons.

Example: when on the homepage (`dspace` theme), the favicon should be the DSpace logo (hardcoded, because the `headTags` section of the `dspace` theme is commented out). After navigating to a community page (`theme-c`, which inherits from `theme-b`, which inherits from `theme-a`), the favicon should switch to ❤️ (configured on `theme-b`).

Make sure this PR works in production mode, when refreshing the page and when using client-side navigation.

When the active theme changes, all elements in the `<head>` section of the DOM that have class `.theme-head-tag` are removed (cf. `.theme-css`). Then any configured elements are added to the `<head>` section.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by paulo-graca** [comment]
I've just tested your code and I'm also getting the heart favicon for communities:
image

**Review by paulo-graca — APPROVED** [review]
I don't have any code suggestions. I managed to have this PR working as expected in my environment (dev and prod). I've also tested your code example and I can confirm that I've got the expected result.

**Comment by tdonohue** [comment]
@bbranan : since this adds an improvement that was requested by you, I wanted to ping you in case you'd be willing to help review/test this PR.

File: src/environments/environment.common.ts
Comment (tdonohue) [inline src/environments/environment.common.ts:296]: This new `headTags` setting needs documentation or inline comments. Either add comments here, or add docs to https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration  (or both).

Currently though, it's not very straightforward what is required in this section, and whether all `<head>` child elements are supported, or just `tagName: link` for now.

**Review by tdonohue — COMMENTED** [review]
👍 @bruno-atmire : Looks good overall & it works.

However, my one request (before we merge this) would be to add some documentation on the new configs...they are not straightforward/self-explanatory (at least not to me).  See my comment inline below.

Once we have comments or docs around those new configurations, I'm OK with merging this.

**Comment by BrunoRoemers** [comment]
@tdonohue and @paulo-graca thanks for the feedback!

I've added some inline comments to `environment.common.ts` and will add the following text to https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-ThemeSettings once I have write access to the wiki. I sent a request for access to wikihelp@lyrasis.org.

### TODO: add the following to wiki under section "Theme Settings"
Each theme has the ability to add a set of (attribute-only) HTML tags in the `<head>` section of the page. This is useful for example to change the **favicon** based on the active theme. Whenever the theme changes, the head tags are reset. A theme can inherit head tags from the parent theme **only** if it doesn't have any head tags itself. (E.g. theme B extends theme A; if theme B does not have head tags, the head tags of theme A will be used (if any). However, if theme B does have head tags, only the tags from theme B will be used.) If none of the themes in the inheritance hierarchy have head tags configured, the head tags of the default theme (if any) will be used. 

Note that a simple hardcoded favicon is set in case no head tags are currently active. The hardcoded favicon is stored at `src/assets/images/favicon.ico`. This implies that if head tags are added to a theme, the favicon should also be configured explicity for that theme, else the behavior is undefined.

**Comment by bbranan** [comment]
Thanks for your work on this @paulo-graca!

For the sake of clarity around the topic raised in the original ticket, I'd suggest that along with what you're already planning to add to the documentation, it would be helpful to also specify exactly how to configure a favicon for a theme. To this end, you might provide an example of the steps and config needed to specify the favicon for someone using the custom theme rather than the default dspace theme. This way no one needs to go digging into the code to understand how to make this work.

**Comment by tdonohue** [comment]
Thanks @bruno-atmire ! The additional inline comments look good to me.  I've also copied your documentation over into the Wiki. I decided to split it into two areas...

* First, added a simple example to the Theme section: https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-ThemeSettings
* Second, added more details about modifying the "headTag" settings to the Customization page here: https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Customization#UserInterfaceCustomization-CustomizeFaviconforsiteortheme

**Review by tdonohue — APPROVED** [review]

**Pull request merged**
